transient-aws/hive-example/dispatch.sh

```diff
@@ -0,0 +1,316 @@
+    echo
+
+    # Iterate on status endpoint for cluster based on response
+
+    echo "Waiting for cluster ${CLUSTER_NAME} to get to READY stage ..."
+
+    progress="0"
+    while
+        STAGE=$(${WGET} -q "${CLUSTER_URL}/status" | jq -r ".stage");
+        [ "${STAGE}" != "READY" ];
+    do
+        if [ "${STAGE}" == "BOOTSTRAP_FAILED" ]; then
+            echo "Cluster ${CLUSTER_NAME} failed to  bootstrap. Please check server logs."
+            exit -1
+        fi
+        if [[ -z "${STAGE}" ]]; then
+            echo "Unable to retrieve status for cluster ${CLUSTER_NAME}. Please check server logs."
+            exit -1
+        fi
+        echo "Stage is ${STAGE}. Progress: ${progress}%..."
+        if [ "$progress" -lt "100" ]; then
+           let "progress+=1"
+        fi        
+        sleep 8
+    done
+
+    echo "Cluster ${CLUSTER_NAME} is provisioned successfully"
+    echo "Cluster ${CLUSTER_NAME} is ready to accept jobs"
+    echo
+fi
+
+# Check the existence of the cluster
+${WGET} -q "${CLUSTER_URL}/status" || { echo "Cluster ${CLUSTER_NAME} not found" >&2; exit -1; }
+
+SUBMIT_JOB="false"
+if [[ -n "$@" ]]; then
+    SUBMIT_JOB="true"
+    # All the arguments are expected to be valid file paths
+    for ARG in "$@"
+    do
+        if ! [ -f "${ARG}" ]; then
+            echo "File '${ARG}' not found"
+            usage
+            exit -2
+        fi
+    done
+fi
+
+if [[ "${SUBMIT_JOB}" == "true" ]]; then
+    # Submit the job
+    # Find the IP address for a gateway
+    echo "Submitting a new job"
+    GATEWAY_IP=$(${WGET} -q "${CLUSTER_URL}" \
+            | jq -r ".instances | map(select(.virtualInstance.template.name == \"${GATEWAY_GROUP_NAME}\"))[0] | .ipAddress")
+
+    echo "Using instance with IP ${GATEWAY_IP} as a gateway to run job ..."
+
+    # Copy all the local files referenced on the command line
+
+
+    JOB_ID="j-${RANDOM_TOKEN}"
+    echo "Dispatching job: ${JOB_ID}"
+
+    REMOTE="${SSH_USERNAME}@${GATEWAY_IP}"
+    SSH_PRIVATE_KEY="${SSH_PRIVATE_KEY/#\~/$HOME}"
+
+    SSH="ssh -o StrictHostKeyChecking=no -i ${SSH_PRIVATE_KEY} ${REMOTE}"
+    SCP="scp -o StrictHostKeyChecking=no -i ${SSH_PRIVATE_KEY}"
+
+    JOB_DIR="/tmp/${JOB_ID}"
+    ${SSH} mkdir ${JOB_DIR}
+
+    for PART in "$@"
+    do
+        ${SCP} ${PART} ${REMOTE}:${JOB_DIR}/
+    done
+
+    JOB_SCRIPT=$(basename $1)
+    shift
+
+    #
+    # Run the job script remotely. The expectation is that the script will block until all the work is done
+    #
+    echo "Starting job: ${JOB_ID}"
+    #${SSH} -t sudo sh -e "${JOB_DIR}/${JOB_SCRIPT}"
+    ${SSH} -t "cd ${JOB_DIR}; chmod +x ${JOB_SCRIPT}; ./${JOB_SCRIPT}"    
+    mkdir -p logs
+    mkdir -p logs/${JOB_ID}
+    ${SCP} ${REMOTE}:/tmp/${SSH_USERNAME}/hive.log ./logs/${JOB_ID}/hive.log
+    # REPLACE_ME with S3 log location
+    aws s3 cp logs/${JOB_ID}/hive.log s3://bucket-name[REPLACE_ME]/etl_jobs/${JOB_ID}/hive.log
```

Comment:
Yes that would be nicer.  For now  I have it as an instruction in the README but we can update this later to take an S3 location for the logs through a script option, and if the option is not passed then we skip this step altogether.